public/app/core/utils/explore.ts

```diff
@@ -67,8 +67,12 @@ export async function getExploreUrl(args: GetExploreUrlArguments): Promise<strin
 
   /** In Explore, we don't have legend formatter and we don't want to keep
    * legend formatting as we can't change it
+   *
+   * We also don't have expressions, so filter those out
    */
-  let exploreTargets: DataQuery[] = panel.targets.map((t) => omit(t, 'legendFormat'));
+  let exploreTargets: DataQuery[] = panel.targets
+    .map((t) => omit(t, 'legendFormat'))
+    .filter((t) => t.datasource?.uid !== '__expr__');
```

Comment:
I couldnt find a good source for this UID for expressions, let me know if there's one somewhere I should use instead of hardcoding it